Source/DataSources/KmlDataSource.js

```diff
@@ -1111,10 +1111,6 @@ define([
             entity.billboard = billboard;
         }
 
-        if (!defined(billboard.image)) {
```

Comment:
By doing this, it fixes your case, but breaks the case where there isn't an icon style at all. In that case Google Earth does show a yellow push pin.

Changing the if to something like
```
if (!defined(styleEntity.billboard) && !defined(billboard.image)) {
```
will only set the default image if an icon style wasn't defined, which I believe is the correct behavior in both cases.